### PR TITLE
Update Lightning Lite docs (2/n)

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Updates the intro.
- Replaces the outdated GIF with an easy to read diff.
- Lists the features of Fabric
- Updates the "Convert to Fabric" section

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source-pytorch/fabric/fabric.rst [hunk 1]
```diff
@@ -3,125 +3,100 @@ Fabric
...
-                loss = model(batch)
-                loss.backward()
-                optimizer.step()
+With only a few changes to your code, Fabric enables for you:
 
+- Automatic placement of models and data onto the device
+- Automatic support for mixed precision (smaller memory footprint)
+- Seamless switching between hardware (CPU, GPU, TPU) and distributed training strategies
+  (data-parallel, model-parallel, etc.)
+- Automated spawning of processes, no launch utilities required
```
Comment (justusschock) [inline docs/source-pytorch/fabric/fabric.rst:52]: maybe state that this is still compatible with launching utilities like torchelastic? This reads as they wouldn't be supported...

File: docs/source-pytorch/fabric/fabric.rst [hunk 2]
```diff
@@ -3,125 +3,100 @@ Fabric
...
 
-:class:`~lightning_fabric.fabric.Fabric` enables pure PyTorch users to scale their existing code
-on any kind of device while retaining full control over their own loops and optimization logic.
+:class:`~lightning_fabric.fabric.Fabric` enables PyTorch users to accelerate and scale their models while retaining full control over their training loop and optimization logic.
 
-.. image:: https://pl-public-data.s3.amazonaws.com/docs/static/images/lite/lightning_lite.gif
-    :alt: Animation showing how to convert your PyTorch code to Fabric.
-    :width: 500
-    :align: center
+.. code-block:: diff
```
Comment (justusschock) [inline docs/source-pytorch/fabric/fabric.rst:8]: TIL! Does this add color-coding for additions and deletions?

File: docs/source-pytorch/fabric/fabric.rst [hunk 3]
```diff
@@ -414,9 +389,9 @@ To define your own behavior, subclass the relevant class and pass it in. Here's
 ----------
 
 
-************************
-Lightning Fabric Methods
-************************
+**************
+Fabric Methods
+**************
```
Comment (justusschock) [inline docs/source-pytorch/fabric/fabric.rst:394]: here you may also need to add the separate methods for setting up only optimizer or only the model.

**Review by justusschock — APPROVED** [review]

File: docs/source-pytorch/fabric/fabric.rst (hunk 3 above)
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst:394]: Agreed, these will have to be added. PRs 3/n etc. will go section by section

File: docs/source-pytorch/fabric/fabric.rst (hunk 2 above)
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst:8]: Yes, it looks like this:

image

**Review by edenlightning — APPROVED** [review]
Nice work!

File: docs/source-pytorch/fabric/fabric.rst [hunk 4]
```diff
@@ -3,125 +3,100 @@ Fabric
 ######
 
 
-:class:`~lightning_fabric.fabric.Fabric` enables pure PyTorch users to scale their existing code
-on any kind of device while retaining full control over their own loops and optimization logic.
+:class:`~lightning_fabric.fabric.Fabric` enables PyTorch users to accelerate and scale their models while retaining full control over their training loop and optimization logic.
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst]: After this change, it is no longer a sentence

File: docs/source-pytorch/fabric/fabric.rst [hunk 5]
```diff
@@ -3,125 +3,100 @@ Fabric
...
 
-        model.train()
-        for epoch in range(args.num_epochs):
-            for batch in dataloader:
-                batch = batch.to(device)
-                optimizer.zero_grad()
-                loss = model(batch)
-                loss.backward()
-                optimizer.step()
+With only a few changes to your code, Fabric enables for you:
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst]: With this change, the bullet points need to continue the sentence from here.

File: docs/source-pytorch/fabric/fabric.rst [hunk 6]
```diff
@@ -3,125 +3,100 @@ Fabric
...
 
+- Automatic placement of models and data onto the device
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst]: That's something fabric does for you so you don't have to do it. I think PyTorch users will very well understand what is meant here. We assume people know PyTorch.

File: docs/source-pytorch/fabric/fabric.rst [hunk 7]
```diff
@@ -3,125 +3,100 @@ Fabric
...
 
 Here are five easy steps to let :class:`~lightning_fabric.fabric.Fabric` scale your PyTorch models.
 
-1. Create the :class:`~lightning_fabric.fabric.Fabric` object at the beginning of your training code.
-2. Remove all ``.to`` and ``.cuda`` calls since :class:`~lightning_fabric.fabric.Fabric` will take care of it.
-3. Apply :meth:`~lightning_fabric.fabric.Fabric.setup` over each model and optimizers pair and :meth:`~lightning_fabric.fabric.Fabric.setup_dataloaders` on all your dataloaders and replace ``loss.backward()`` by ``fabric.backward(loss)``.
-4. Run the script from the terminal using ``lightning run model path/to/train.py`` or use the :meth:`~lightning_fabric.fabric.Fabric.launch` method in a notebook.
-
-|
+**Step 1:** Create the :class:`~lightning_fabric.fabric.Fabric` object at the beginning of your training code.
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst:66]: Added it in the code here. Don't think it is worth it is own step.
